network/p2p/node_sampler.go

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+// NodeSampler samples nodes in network
+type NodeSampler interface {
+	// Sample returns at most [limit] nodes. This may return fewer nodes if
+	// fewer than [limit] are available.
+	Sample(ctx context.Context, limit int) []ids.NodeID
```

Comment:
note: I also considered making another type called:
```
type Node struct {
    ID ids.NodeID
}
```

so we could add onto it without causing breaking changes. I felt like it was reasonable that in the future we might add some peer-level metadata to this, but it seemed premature to add for now.